assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { render, findByLabelText, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import FrontendBlock from '../frontend';
+
+describe( 'FrontendBlock', () => {
+	let validationData = {
+		hasValidationErrors: false,
+		getValidationError: jest.fn(),
+		clearValidationError: jest.fn(),
+		hideValidationError: jest.fn(),
+		setValidationErrors: jest.fn(),
+	};
+	beforeEach( () => {
+		validationData = {
+			hasValidationErrors: false,
+			getValidationError: jest.fn(),
+			clearValidationError: jest.fn(),
+			hideValidationError: jest.fn(),
+			setValidationErrors: jest.fn(),
+		};
+	} );
+
+	it( "Renders a checkbox if the checkbox prop is passed and doesn't if it's not.", async () => {
+		const { container, rerender } = render(
+			<FrontendBlock
+				checkbox={ true }
+				text={ 'I agree to the terms and conditions' }
+				validation={ validationData }
+			/>
+		);
+
+		const checkbox = await findByLabelText(
+			container,
+			'I agree to the terms and conditions'
+		);
+
+		expect( checkbox ).toBeTruthy();
+		expect( checkbox.getAttribute( 'type' ) ).toEqual( 'checkbox' );
+
+		rerender(
+			<FrontendBlock
+				checkbox={ true }
+				text={ 'I agree to the terms and conditions' }
+				validation={ validationData }
+			/>
+		);
+		const textOnly = await screen.findByLabelText(
```

Comment:
Why are we using `screen.findByLabelText` here and just `findByLabelText` above? Just checking this is intended

Reply:
My bad, removed!